### PR TITLE
Replace IDataMigrationManager.UpdateAsync(string) with IDataMigrationManager.UpdateAsync(IEnumerable<string>)

### LINKED ISSUE #4917: Move IDataMigrationManager.UpdateAsync() to an extension method
While i'm working on data docs, I notice that `IDataMigrationManager` has two versions of `UpdateAsync()` one accept feature and the other one accepts a list of features

IMHO the `UpdateAsync(IEumerable<string> features)` could be an extension methods instead

/cc @jtkech

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #4917

No breaking changes, but we should remove the obsolete method before a major release

### CODE REVIEW

**Review by Piedone — COMMENTED** [review]
This should be the other way around: https://github.com/OrchardCMS/OrchardCore/issues/4917#issuecomment-2081669456

**Review by Piedone — CHANGES_REQUESTED** [review]
This should be the other way around: https://github.com/OrchardCMS/OrchardCore/issues/4917#issuecomment-2081669456

**Comment by hishamco** [comment]
Do you mean leave `UpdateAsync(IEnumerable<string>)` and remove `UpdateAsync()`? if yes I might go with `params string[]` instead, this will support both

File: src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigrationManager.cs [hunk 1]
```diff
@@ -28,7 +30,8 @@ public interface IDataMigrationManager
         /// Updates the database to the latest version for the specified features.
         /// </summary>
         /// <param name="features">The features to be updated.</param>
-        Task UpdateAsync(IEnumerable<string> features);
+        [Obsolete("This method has been deprecated and moved into an extension method.")]
+        async Task UpdateAsync(IEnumerable<string> features) => await Task.WhenAll(features.Select(UpdateAsync));
```
Comment (MikeAlhayek) [inline src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigrationManager.cs:34]: Do not use Task.WhenAll() for the same reason @Piedone mention above.

Comment (hishamco) [reply src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigrationManager.cs:34]: So, how will they be optimized then?

**Comment by Piedone** [comment]
> Do you mean leave `UpdateAsync(IEnumerable<string>)` and remove `UpdateAsync()`? if yes I might go with `params string[]` instead, this will support both

Yes.

File: src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigrationManager.cs (hunk 1 above)
Comment (Piedone) [reply src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigrationManager.cs:34]: Not with parallel SQL queries :).

**Comment by Piedone** [comment]
How did you test this?

**Comment by hishamco** [comment]
If you look closely nothing changes because the `UpdateAsync(string)` calling `UpdateAsync(string[])`, both are well tested before, but I can do manual testing quickly

**Comment by Piedone** [comment]
Please, and I've asked for this multiple times before, test your code. Don't ask for a review before that. The change might be trivial, or it might seem trivial but actually break something, but testing your code is step 0 of QA. You're not finished until you've run your code and are sure that it does what it should. This is your responsibility as the PR author.

**Comment by hishamco** [comment]
> This is your responsibility as the PR author.

I know what you mean Zoltan, but if you look at the change I didn't add anything fancy, just deprecate the `UpdateAsync(string)`, anyway I tested both, and everything working as expected

FYI no unit tests were added here, I just did a manual test

**Comment by Piedone** [comment]
And _how_ did you do the manual test? I'm asking this because I'd like to try that too.

**Comment by hishamco** [comment]
1. Create few migrations (even empty)
2. Create a razor page that injects `IDataMigrationManager` and call `UpdateAsync(string)`
3. Run the app and access the created page
4. Check your migrations record in database

Try the above step with `UpdateAsync(string[])` and don't forgot to disable the `AutomaticDataMigration` registration before the test to ensure your code is the only place to update the migration

File: src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigrationManager.cs [hunk 2]
```diff
@@ -22,13 +23,14 @@ public interface IDataMigrationManager
         /// Updates the database to the latest version for the specified feature.
         /// </summary>
         /// <param name="feature">The feature to be uninstalled.</param>
-        Task UpdateAsync(string feature);
+        [Obsolete("This method has been deprectated please use UpdateAsync(string[] features) instead.")]
```
Comment (Piedone) [inline src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigrationManager.cs:26]:
```suggestion
        [Obsolete("This method has been deprecated, please use UpdateAsync(string[] features) instead.")]
```

File: src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigrationManager.cs [hunk 3]
```diff
@@ -22,13 +23,14 @@ public interface IDataMigrationManager
...
+        Task UpdateAsync(string feature) => UpdateAsync([feature]);
 
         /// <summary>
-        /// Updates the database to the latest version for the specified features.
+        /// Updates the database to the latest version for the specified feature(s).
         /// </summary>
-        /// <param name="features">The features to be updated.</param>
-        Task UpdateAsync(IEnumerable<string> features);
+        /// <param name="features">The feature(s) to be updated.</param>
+        Task UpdateAsync(params string[] features);
```
Comment (Piedone) [inline src/OrchardCore/OrchardCore.Data.Abstractions/IDataMigrationManager.cs:33]: FYI this is a breaking change, but it's not something that anybody should actually notice.

**Review by Piedone — CHANGES_REQUESTED** [review]

**Review by Piedone — APPROVED** [review]

**Pull request merged**